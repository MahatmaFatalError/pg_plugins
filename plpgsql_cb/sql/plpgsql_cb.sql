--
-- Tests for PL/pgSQL instrumentation callbacks.
--

LOAD 'plpgsql_cb';

CREATE OR REPLACE FUNCTION public.f1()
RETURNS void
LANGUAGE plpgsql
AS $$
DECLARE x int;
BEGIN
  x := 2;
  RETURN;
END;
$$;

CREATE OR REPLACE FUNCTION public.f2()
RETURNS void
LANGUAGE plpgsql
AS $$
DECLARE x int;
BEGIN
  BEGIN
    x := 3;
    RETURN;
  END;
END
$$;

SELECT f1();
SELECT f2();
