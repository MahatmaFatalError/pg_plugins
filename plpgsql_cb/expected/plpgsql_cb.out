--
-- Tests for PL/pgSQL instrumentation callbacks.
--
LOAD 'plpgsql_cb';
CREATE OR REPLACE FUNCTION public.f1()
RETURNS void
LANGUAGE plpgsql
AS $$
DECLARE x int;
BEGIN
  x := 2;
  RETURN;
END;
$$;
CREATE OR REPLACE FUNCTION public.f2()
RETURNS void
LANGUAGE plpgsql
AS $$
DECLARE x int;
BEGIN
  BEGIN
    x := 3;
    RETURN;
  END;
END
$$;
SELECT f1();
NOTICE:  function setup: "f1()"
NOTICE:  function beg: "f1()"
NOTICE:  statement beg - ln: 4
NOTICE:  statement end - ln: 4
NOTICE:  statement beg - ln: 5
NOTICE:  statement end - ln: 5
NOTICE:  function end: "f1()"
 f1 
----
 
(1 row)

SELECT f2();
NOTICE:  function setup: "f2()"
NOTICE:  function beg: "f2()"
NOTICE:  statement beg - ln: 4
NOTICE:  statement beg - ln: 5
NOTICE:  statement end - ln: 5
NOTICE:  statement beg - ln: 6
NOTICE:  statement end - ln: 6
NOTICE:  statement end - ln: 4
NOTICE:  function end: "f2()"
 f2 
----
 
(1 row)

